File: 프로그래머스/기초 트레이닝 - Level 0/접두사인지 확인하기-solve.py
"""
어떤 문자열에 대해 접두사는 특정 인덱스까지 문자열 의미 

is_prefix길이만큼 반복하면서 is_prefix 문자와 my_string문자가 같은지 확인 
is_prefix길이가 더 길면 접두사가 아님
"""

def solution(my_string, is_prefix):
    answer = 1
    
    if len(is_prefix) > len(my_string):
        return 0  
    
    for i in range(len(is_prefix)):
        if my_string[i] != is_prefix[i]:
            answer = 0 
            
    return answer

File: 프로그래머스/기초 트레이닝 - Level 0/접두사인지 확인하기-solve.java
class Solution {
    public int solution(String my_string, String is_prefix) {
        int answer = 1;
        int length = is_prefix.length(); 
        
        if (is_prefix.length() > my_string.length()) {
            return 0;
        }
        
        for (int i = 0; i < is_prefix.length(); i++) {
            if (is_prefix.charAt(i) != my_string.charAt(i)) {
                answer = 0;
            }
        }
        
        return answer;
    }
}